spark-3.1-spanner-lib/src/main/java/com/google/cloud/spark/spanner/SpannerRelation.java

```diff
@@ -38,6 +38,19 @@ public SpannerRelation(String table, SparkSession session) {
     this.session = session;
   }
 
+  private String buildSQL(String[] requiredColumns, Filter[] filters) {
```

Comment:
This can be static, as it does not use any members. Also, please make it package protected, and add unit test for the various cases